orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -485,10 +446,15 @@ def _get_inference_graph(self, *inputs, **kwargs):
         '''
 
         # Setup dynamic axes for onnx model
-        input_names, dynamic_axes, self._input_names_require_grad, _ = _parse_inputs_for_onnx_export(self._original_module_input_names, None, *inputs, **kwargs)
-        output_names, output_dynamic_axes, self._original_module_output_type = _parse_outputs_for_onnx_export(self._original_module, inputs, kwargs)
+        input_names, dynamic_axes, self._input_names_require_grad, _ = \
+            _parse_inputs_for_onnx_export(self._original_module_input_names, None, *inputs, **kwargs)
+        output_names, output_dynamic_axes, self._original_module_output_schema = \
+            _parse_outputs_for_onnx_export_and_extract_output_schema(self._original_module, inputs, kwargs)
         dynamic_axes.update(output_dynamic_axes)
 
+        # Get the module that flattens the output from the original module into a tuple
+        self._flattened_output_module = _ortmodule_output_transformation.get_flattened_output_module(self._original_module)
```

Comment:
Any reason to not do this during `ORTModule.__init__`? It would simplify `eval` and `training` as `_flattened_output_module  will be valid from the beginning

Reply:
No, reason in particular I guess except that it was needed for export. I will move it to ```__init__```.